File: packages/jexlang-java/src/main/java/com/jexlang/java/eval/errors/JexLangRuntimeError.java
package com.jexlang.java.eval.errors;

public class JexLangRuntimeError extends Error {
    private final String name;

    public JexLangRuntimeError(String message) {
        super(message);
        this.name = "JexLangRuntimeError";
    }

    public String getName() {
        return name;
    }
}

File: packages/jexlang-swift/jexlang-swift/eval/errors/JexLangRuntimeError.swift
//
//  JexLangRuntimeError.swift
//  jexlang-swift
//
//  Created by Vinay Kumar on 03/10/25.
//

import Foundation

public class JexLangRuntimeError: Error {
    private let name: String
    private let message: String
    
    public init(message: String) {
        self.message = message
        self.name = "JexLangRuntimeError"
    }
    
    public func getName() -> String {
        return name
    }
}

extension JexLangRuntimeError: LocalizedError {
    public var errorDescription: String? {
        return message
    }
    
    public var failureReason: String? {
        return name
    }
}

extension JexLangRuntimeError: CustomStringConvertible {
    public var description: String {
        return "\(name): \(message)"
    }
}
